specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,503 @@
+  HRESULT GetProcessInfosWithDetails([in] ICoreWebView2GetProcessInfosWithDetailsCompletedHandler* handler);
+}
+
+/// A continuation of the ICoreWebView2FrameInfo interface.
+[uuid(a7a7e150-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2FrameInfo2 : ICoreWebView2FrameInfo {
+  /// This frame's parent frame's `FrameInfo`. `ParentFrameInfo` will only be
+  /// populated when obtained via calling 
+  /// `CoreWebView2ProcessInfo.AssociatedFrameInfos`. 
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed` will
+  /// always have a `null` `ParentFrameInfo`. This property is also `null` for the  
+  /// top most document in the WebView2 which has no parent frame.
+  /// Note that this `ParentFrameInfo` could be out of date as it's a snapshot.   
+  [propget] HRESULT ParentFrameInfo([out, retval] ICoreWebView2FrameInfo** frameInfo);
+  /// The unique identifier of the frame associated with the current `FrameInfo`.
+  /// It's the same kind of ID as with the `FrameId` in `CoreWebView2` and via 
+  /// `CoreWebView2Frame`. `FrameId` will only be populated when obtained
+  /// calling `CoreWebView2ProcessInfo.AssociatedFrameInfos`. 
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed` will
+  /// always have an invalid frame Id 0.
+  /// Note that this `FrameId` could be out of date as it's a snapshot. 
+  [propget] HRESULT FrameId([out, retval] UINT32* id);
+  /// The frame kind of the frame. `FrameKind` will only be populated when
+  /// obtained calling `CoreWebView2ProcessInfo.AssociatedFrameInfos`.
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed`
+  /// will always have the default value `COREWEBVIEW2_FRAME_KIND_OTHER`.
+  /// Note that this `FrameKind` could be out of date as it's a snapshot.
+  [propget] HRESULT FrameKind([out, retval] COREWEBVIEW2_FRAME_KIND* kind);
+}
+
+/// A continuation of the ICoreWebView2Frame4 interface.
+[uuid(04baa798-a0e9-11ed-a8fc-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2Frame5: ICoreWebView2Frame4 {
+  /// The unique identifier of the current frame. It's the same kind of ID as with 
+  /// the `FrameId` in `CoreWebView2` and via `CoreWebView2FrameInfo`.
+  [propget] HRESULT FrameId([out, retval] UINT32* id);
+}
+
+/// A continuation of the ICoreWebView2_17 interface.
+[uuid(ad712504-a66d-11ed-afa1-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2_18 : ICoreWebView2_17 {
+  /// The unique identifier of the current frame. It's the same kind of ID as with 
+  /// the `FrameId` in `CoreWebView2Frame` and via `CoreWebView2FrameInfo`.
+  /// Note that `FrameId` may not be valid if `CoreWebView` has not done
+  /// any navigation. It's safe to get this value during or after the first
+  /// `ContentLoading` event. Otherwise, it could return the invalid frame Id 0.
+  [propget] HRESULT FrameId([out, retval] UINT32* id);

```

Comment:
Will developers want the Frame instead of FrameId value?  How often will developers want to use this id to lookup a CoreWebView2Frame, or is the common case to log this for diagnostics?

Reply:
For memory monitoring cases, we expect developers not only want to acquire resource consumption for each frame tree, but also have chance to take further operations for the corresponding CoreWebView2/CoreWebView2Frame. Not sure how often it is, but it's a requirement for our existing customers.

Reply:
No change: We will want to do this in the future but its complicated and not required for this spec now.